anchors/anchor_repository_integration_test.go

```diff
@@ -31,13 +33,68 @@ func TestMain(m *testing.M) {
 	os.Exit(result)
 }
 
-func TestCommitAnchor_Integration(t *testing.T) {
+func TestPreCommitAnchor_Integration(t *testing.T) {
 	anchorID := utils.RandomSlice(32)
+	signingRoot := utils.RandomSlice(32)
+
+	anchorIDTyped, _ := anchors.ToAnchorID(anchorID)
+	preCommitAnchor(t, anchorID, signingRoot)
+	valid := anchorRepo.HasValidPreCommit(anchorIDTyped)
+	assert.True(t, valid)
+}
+
+func TestPreCommit_CommitAnchor_Integration(t *testing.T) {
+	anchorIDPreImage := utils.RandomSlice(32)
+	h := sha256.New()
+	_, err := h.Write(anchorIDPreImage)
+	assert.NoError(t, err)
+	var anchorID []byte
+	anchorID = h.Sum(anchorID)
+	proofStr := []string{"0x73bb733279cd232d72732afad693f80510e71262738b1205a061a1c34497e49c", "0x408b2caa80ace6ac3a37be957235011c0053f0a561f5a8dcf66d223bfffccecb"}
+	signingRootStr := "0x6ae9e6cc91cded82896d2439942fd42412b5a2ff5fd45bbed0f5a20de0b962c2"
+	documentRootStr := "0xf5c8f866f4acf2e2e74a803f86cd2a7ac9285721259b172ef121417e886ca22a"
+
+	signingRoot, err := hexutil.Decode(signingRootStr)
+	assert.NoError(t, err)
+
+	documentRoot, err := hexutil.Decode(documentRootStr)
+	assert.NoError(t, err)
+
+	proof1, err := hexutil.Decode(proofStr[0])
+	assert.NoError(t, err)
+
+	proof2, err := hexutil.Decode(proofStr[1])
+	assert.NoError(t, err)
+
+	var proofB1 [32]byte
+	copy(proofB1[:], proof1)
+	var proofB2 [32]byte
+	copy(proofB2[:], proof2)
+
+	anchorIDTyped, _ := anchors.ToAnchorID(anchorID)
+	preCommitAnchor(t, anchorID, signingRoot)
+	valid := anchorRepo.HasValidPreCommit(anchorIDTyped)
+	assert.True(t, valid)
+
+	docRootTyped, _ := anchors.ToDocumentRoot(documentRoot)
+	commitAnchor(t, anchorIDPreImage, documentRoot, [][anchors.DocumentProofLength]byte{proofB1, proofB2})
+	gotDocRoot, err := anchorRepo.GetDocumentRootOf(anchorIDTyped)
+	assert.Nil(t, err)
+	assert.Equal(t, docRootTyped, gotDocRoot)
+}
+
+func TestCommitAnchor_Integration(t *testing.T) {
+	anchorIDPreImage := utils.RandomSlice(32)
+	h := sha256.New()
+	_, err := h.Write(anchorIDPreImage)
+	assert.NoError(t, err)
+	var anchorID []byte
+	anchorID = h.Sum(anchorID)
```

Comment:
```suggestion
	anchorID = h.Sum(nil)
```

Reply:
I think its better the way that it is right now, as the value is in callers control